### PR TITLE
feat: add github project api

### LINKED ISSUE #248: Use GitHub API to get Project Data

### PR DESCRIPTION
**Pull request by schmelto**
resolve #248 

````js
let githubprojects = [
  {
    name: 'chryz-hub/chryz-hub.github.io',
    icon: 'description',
  },
  {
    name: 'chryz-hub/chryz-hub.github.io',
    icon: 'article',
  },
  {
    name: 'chryz-hub/chryz-hub.github.io',
    icon: 'person',
  },
  // Here can be added some more projets if needed
];
```

### CODE REVIEW

**Comment by namangupta1399** [comment]
Looks good 👍

File: src/assets/css/styles.css [hunk 1]
```diff
@@ -1,262 +1,277 @@
...
 .project div span {
-    margin: 0 5px;
-    color: var(--secondaryBtn);
-}
\ No newline at end of file
+  margin: 0 5px;
+  color: var(--secondaryBtn);
+}
+
+.flex-col-2 {
```
Comment (schmelto) [inline src/assets/css/styles.css:265]: I use css columns to make the project cards responsive (#247)

File: index.html [hunk 2]
```diff
@@ -75,25 +84,38 @@ <h1 class="text-primary">Chryz-Hub</h1>
...
+
     <section id="projects" class="projects">
       <h2>Projects</h2>
-      <div class="flex-items">
+
+      <div class="flex-col-2" id="githubprojects">
+        <!-- GitHub Projects from scripts.js goes here -->
+      </div>
+
+      <!-- Not needed if we go to api call in the js -->
```
Comment (schmelto) [inline index.html:112]: Project cards are not needed in the index html due we edit the html dom direktly in the scrpit

File: src/assets/css/styles.css (hunk 1 above)
Comment (namangupta1399) [reply src/assets/css/styles.css:265]: Yeah, this is nice. Actually, I've not used it much. That's why didn't came with this approach.
I will remove my commit (on responsiveness) and we can go ahead with your changes

File: index.html (hunk 2 above)
Comment (namangupta1399) [reply index.html:112]: Right, we just created them for reference. We can remove them once they are added through the script.

Comment (schmelto) [reply index.html:112]: should we go with this project card design?

Comment (namangupta1399) [reply index.html:112]: That's fine. But, adding bootstrap now would add unnecessary styling. Wouldn't it?

**Comment by schmelto** [comment]
I think now its on a level you can review it.
There is a lot more to do but I think other folks can handle this

File: src/assets/javascript/script.js [hunk 3]
```diff
@@ -1,23 +1,75 @@
...
+    name: 'chryz-hub/chryz-hub.github.io',
+  },
+  {
+    name: 'chryz-hub/chryz-hub.github.io',
+  },
+  // Here can be added some more projets if needed
+];
+
+githubprojects.forEach((project) => {
+  getproject(project.name);
```
Comment (schmelto) [inline src/assets/javascript/script.js:43]: maybe we can discuss the method naming convention here due getproject could be a little bit missleading

Comment (namangupta1399) [reply src/assets/javascript/script.js:43]: Lol, I merged at the same time 🤣

File: src/assets/javascript/script.js [hunk 4]
```diff
@@ -1,23 +1,75 @@
...
+  // removing github emojis cause they are not usable in html
+  project.description = project.description.replace(/:[^}]*:/, '');
+
+  // TODO: add Project Card
+  let projectcard = `<a href="${project.html_url}">
+    <div class="item project">
+      <h4 class="text-secondary">${project.name}</h4>
+      <p>${project.description}</p>
+      <div>
+      <!-- TODO add icons -->
```
Comment (schmelto) [inline src/assets/javascript/script.js:67]: This one I would leave for someone lese

File: src/assets/javascript/script.js (hunk 3 above)
Comment (namangupta1399) [reply src/assets/javascript/script.js:43]: Yeah, we can work on the naming convention

File: src/assets/css/styles.css [hunk 5]
```diff
@@ -1,262 +1,277 @@
...
+  width: 100%;
+  height: auto;
+  columns: 2;
```
Comment (schmelto) [inline src/assets/css/styles.css:268]: We could discuss if we want to have more columns

Comment (namangupta1399) [reply src/assets/css/styles.css:268]: Yeah, sure 👍

File: src/assets/javascript/script.js (hunk 4 above)
Comment (schmelto) [reply src/assets/javascript/script.js:67]: it was not that much :D

**Comment by Evavic44** [comment]
Nice update so far guys. Sorry I have not been active.

**Pull request merged**